js_modules/dagster-ui/packages/ui-core/src/hooks/useThrottledMemo.tsx

```diff
@@ -0,0 +1,71 @@
+import {createContext, useContext, useEffect, useRef, useState} from 'react';
+import {unstable_batchedUpdates} from 'react-dom';
+
+/**
+ * Context to accumulate updates over 200ms and batch them together to reduce the number of renders
+ */
+export const ThrottledMemoBatchingContext = createContext<{enqueue: (update: () => void) => void}>(
+  (() => {
+    let isScheduled = false;
+    const queue: Array<() => void> = [];
+    return {
+      enqueue: (update: () => void) => {
+        queue.push(update);
+        if (!isScheduled) {
+          isScheduled = true;
+          setTimeout(() => {
+            unstable_batchedUpdates(() => {
+              while (queue.length) {
+                queue.shift()!();
+              }
+              isScheduled = false;
+            });
+          }, 200);
+        }
+      },
+    };
+  })(),
+);
```

Comment:
This is so I can override it in tests, but yeah for my particular tests I didn't end up needing it. Basically trying to avoid having to add more __resetForJest type functions that I kept needing to add with other hooks relying on global state.